### ISSUE TITLE
Please add or fork repo for Upgradeable Diamond Storage enabled contracts

### ISSUE THREAD

**Issue by GVTopCoder** [OpenZeppelin/openzeppelin-contracts#3131]
**🧐 Motivation**
Support the EIP2535 Standard for upgradeable contracts.  

**📝 Details**
This is a transpiler upgrade to generate Upgradeable EIP2535 Diamond Storage enabled contracts.  This repo is generated just as the openzeppelin-contracts-upgradable one is with transpiler converting to EIP2535 standard that uses storage slots instead of gaps for storage.  This is still a WIP, but I plan on having it finished in the next couple of days.

I would like to have a fork repo created under the OpenZeppelin Organization to push the new repo back into.  But if not feasible, I can keep the current repo URL: https://github.com/GeniusVentures/openzeppelin-contracts-diamond.

A pull request has already been submitted to the transpiler repo as well, but it is probably better pulled when completed.

**Comment by Amxx** [comment]
Hello @GVTopCoder 

2535 is definitelly in our radar, and we even have a implementation of the proxy somewhere, but we indeed believe it would need a specific "module-based" implementation of our contracts. It would also need dedicated tooling to manage it securely.

We don't have the bandwidth to handle that now, but I'd definitely have a look at any proposal. I'm not sure it can be done through the transpiler alone ... but maybe it can.

**Comment by frangio** [comment]
This is a duplicate of https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2964.

I still feel like I shared in [my comments there](https://github.com/OpenZeppelin/openzeppelin-contracts/issues/2964#issuecomment-972861949):

>I think this is actually a very good idea for the upgradeable contracts, but we will need to carefully consider it for the next major release because it is a breaking change.

We don't really want to maintain a new variant of the contracts, maintaining two is already significant workload, so if we go this route we would make it the default Upgradeable variant.

It's interesting that you've done the work on the transpiler, though like I commented in [your PR](https://github.com/OpenZeppelin/openzeppelin-transpiler/pull/69) I'm missing a more detailed description of the changes.

Are you planning to use this for your project?
